lib/Runtime/ByteCode/ByteCodeGenerator.cpp

```diff
@@ -4984,8 +4984,8 @@ void AssignRegisters(ParseNode *pnode, ByteCodeGenerator *byteCodeGenerator)
                 }
             }
 
-            // An eval call in a class member needs to load super.
-            if (funcInfo->root->sxFnc.IsClassMember())
+            // An eval call in a method or a constructor needs to load super.
+            if (funcInfo->root->sxFnc.IsMethod() || funcInfo->root->sxFnc.IsConstructor())
```

Comment:
No, because IsConstructor = !(IsAsync || IsLambda || IsAccessor). For example, for
function f() {}
IsMethod == FALSE
IsConstructor == TRUE


---
In reply to: [75549267](https://github.com/Microsoft/ChakraCore/pull/1466#discussion_r75549267) [](ancestors = 75549267)